pkg/controllers/status/cluster_status_controller.go

```diff
@@ -376,7 +378,9 @@ func (c *ClusterStatusController) initLeaseController(cluster *clusterv1alpha1.C
 
 		klog.Infof("Stop syncing lease for cluster: %s", cluster.Name)
 		c.ClusterLeaseControllers.Delete(cluster.Name) // ensure the cache is clean
+		wg.Done()
 	}()
+	wg.Wait()
```

Comment:
The lease controller will keep running and never stop in normal cases. Seems we don't need to wait for it to complete.